src/globalComponents/Leaderboard.js

```diff
@@ -22,10 +22,47 @@ class Leaderboard extends React.Component<Props, State> {
 		super();
 
 		this.state = {
-			leaderNames: ["BILL", "JOHN", "gosh", "dang", "bro", "Emil", "Baylor", "Adam", "David", "Jorge"] // NEED AN API CALL HERE
+			leaderNames: [] // NEED AN API CALL HERE
 		}
 	}
 
+	componentDidMount() {
+		this.getLeaders();
+	}
+
+	//This function gets the leaders for the leaderboard
+	getLeaders(): void {
+		const responsePromise: Promise<Response> = fetch('/api/user/leaderboard', {
+			method: 'GET',
+			headers: {
+				'Access-Control-Allow-Origin': '*',
+				'Content-Type': 'application/json',
+				'Access-Control-Allow-Credentials': 'true'
+			},
+		});
+		responsePromise.then(
+			response => response.json().then(data => {
+				if (response.status !== 200) {
+					console.log(response.status);
+					console.log(data.msg);
+					console.log(data);
+				}
+				else {
+					let leaders = [];

```

Comment:
There is a cleaner way of doing this:

`this.setState({leaderNames: data.map(usernames => usernames.userName)`

Reply:
Thanks that was a lot cleaner